### ISSUE TITLE
"Cannot find module 'jspm-git'" message when linking to package from custom registry

### ISSUE THREAD

**Issue by nfriend** [jspm/jspm-cli#1877]
I've defined a custom registry that points to my company's internal GitLab instance:

````
jspm config registries.mycompany.version 1.0
jspm config registries.mycompany.baseurl https://gitlab.mycompany.com/
jspm config registries.mycompany.handler jspm-git
```` 

This registry is working correctly for normal `jspm installs`.  However, if I try and `jspm link` a local package in place of one of the packages hosted on our private git server, I get the following error:

```
PS C:\Users\username\code\my-module> jspm link mycompany:my-project/my-module

err  Error: Cannot find module 'jspm-git'
         at Function.Module._resolveFilename (module.js:339:15)
         at Function.Module._load (module.js:290:25)
         at Module.require (module.js:367:17)
         at require (internal/module.js:16:19)
         at Object.exports.load (C:\Users\username\AppData\Roaming\npm\node_modules\jspm\lib\registry.js:68:25)
         at Object.derivePackageConfig (C:\Users\username\AppData\Roaming\npm\node_modules\jspm\lib\package.js:658:27)
         at C:\Users\username\AppData\Roaming\npm\node_modules\jspm\lib\link.js:58:20
         at lib$rsvp$$internal$$tryCatch (C:\Users\username\AppData\Roaming\npm\node_modules\jspm\node_modules\rsvp\dist\rsvp.js:493:16)
         at lib$rsvp$$internal$$invokeCallback (C:\Users\username\AppData\Roaming\npm\node_modules\jspm\node_modules\rsvp\dist\rsvp.js:505:17)
         at lib$rsvp$$internal$$publish (C:\Users\username\AppData\Roaming\npm\node_modules\jspm\node_modules\rsvp\dist\rsvp.js:476:11)
         at lib$rsvp$asap$$flush (C:\Users\username\AppData\Roaming\npm\node_modules\jspm\node_modules\rsvp\dist\rsvp.js:1198:9)
         at nextTickCallbackWith0Args (node.js:453:9)
         at process._tickCallback (node.js:382:13)

err  Unable to load registry mycompany
```

I've tried every combination of local and global installations for both `jspm` and `jspm-git`:

- local `jspm`, local `jspm-git`
- local `jspm`, global `jspm-git`
- global `jspm`, local `jspm-git`
- global `jspm`, global `jspm-git`

All of these result in the error message above.  If I install `jspm-git` locally, I'm able to `require` it from node:

```
PS C:\Users\username\code\my-module> node
> require('jspm-git');
require('jspm-git');
{ [Function] configure: [Function] }
>
```
But I still get the "cannot find module 'jspm-git'" error when running `jspm link`.

Am I doing something wrong?  Is this an issue with JSPM or should I be posting an issue on the `jspm-git` repository?

**Comment by guybedford** [comment]
I believe this was fixed on the 0.17 branch of jspm with the commit in https://github.com/jspm/jspm-cli/commit/52a6d17f44c635c740491734152664d6626ba117.

It could be possible to try backporting that to 0.16.

**Comment by nfriend** [comment]
Great, thanks for the quick response.  Looking forward to the 0.17 release.
